### PR TITLE
Improve submission links

### PR DESCRIPTION
**Pull request by Jezzamonn**
This is unfinished... I kinda got stuck figuring out the whole state updating thing in preact. I figured I'd make the pull request early so you can see my progress or explain how things work :)

### CODE REVIEW

**Comment by Jezzamonn** [comment]
Alright, I may have got this working. Currently my VM is broken due to the new way the api is handled, so I haven't tested this at all, but if I revert to an older version I should be able to do that.

Also still to be done is how a game displays the links when it's not editing, that shouldn't be very hard though.

**Comment by povrazor** [comment]
New way the API is handled?

**Comment by Jezzamonn** [comment]
I briefly mentioned it on gitter, but basically no api calls were working due to cross domain stuff, so I couldn't test whether it worked.

**Comment by Jezzamonn** [comment]
Ok, figured out what was going wrong.

Now I'm adding the server side stuff. These urls can now be saved, but they don't appear when the page is reloaded. 🤔

**Comment by Jezzamonn** [comment]
Last thing left to do is make it so when you edit the links, the save button changed from "saved" to "save". Otherwise it looks like everything was automatically saved.

However, that button is part of ContentSimple (the edit thing used for posts) and it's state is managed by that ContentSimple's state, so I'm not sure if there's any way to update that.

I think The React Way would be to have that editing stuff dumb and not handle state, and then whenever you want an edit thing you need to add some parent thing that handles state.

**Comment by Jezzamonn** [comment]
An alternative would be to update this stuff as it's edited, same as the other game stuff. Joining it with the save button can happen later, I guess

**Comment by Jezzamonn** [comment]
Ok, should be good to go! Have a look at the final diff because I made some changes to some of the other text on the page too to make it feel more complete

File: src/com/content-item/item.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ export default class ContentItem extends Component {
 		});
 		for (let i = 0; i < 5; i ++) {
 			this.state.linkNames[i] = node.meta['link-0' + (i+1) + '-name'];
```
Comment (povrazor) [inline src/com/content-item/item.js]: // TODO: Make this support more than 9 links

File: src/com/content-item/item.js [hunk 2]
```diff
@@ -66,7 +66,7 @@ export default class ContentItem extends Component {
...
-			this.state.linkUrls[i] = node.meta['link-0' + (i+1) + '-name'];
+			this.state.linkUrls[i] = node.meta['link-0' + (i+1) + '-url'];
```
Comment (povrazor) [inline src/com/content-item/item.js]: Don't use `-url`. Instead, have no suffix. Assume that `link-01` is a URL, and if no target (`B`) is set on the link, that `link-01-name` is how to set a name. When `B` is non-zero, the Tag node's `name` is used instead.

File: public-api/vx/node.php [hunk 3]
```diff
@@ -28,6 +28,17 @@
 		'grade-06-out' => ['length' => 1],
 		'grade-07-out' => ['length' => 1],
 		'grade-08-out' => ['length' => 1],
+		
+		'link-01-name' => ['length' => 64],
+		'link-02-name' => ['length' => 64],
+		'link-03-name' => ['length' => 64],
+		'link-04-name' => ['length' => 64],
+		'link-05-name' => ['length' => 64],
+		'link-01-url' => ['length' => 128],
```
Comment (povrazor) [inline public-api/vx/node.php]: Gosh, I wish we could limit URLs to 128 characters, but it'll have to be longer. I need to do a quick check first, but I'm guessing `2048` will cover us. Also it should be `link-01` without the `-url` (as the `-name` feature will become less and less needed as we populate our platform tag list). ;)

Comment (povrazor) [inline public-api/vx/node.php]: Yeah, go with `2048`.

Comment (pschichtel) [inline public-api/vx/node.php]: https://stackoverflow.com/questions/219569/best-database-field-type-for-a-url/219664#219664

Comment (povrazor) [inline public-api/vx/node.php]: Internally it is `Text`, but this limit is protection to mitigate attacks.

Comment (povrazor) [inline public-api/vx/node.php]: Actually no, lets go with `512`, and wait for someone to complain.

**Comment by Jezzamonn** [comment]
FYI I deleted the showPrePub thing ("Hey folks! We're still finishing the data fields below ..."). Don't think it's needed anymore.

**Comment by povrazor** [comment]
Sounds good.

**Comment by Jezzamonn** [comment]
Yay!

**Pull request merged**